router/batchrouter/asyncdestinationmanager/sftp/manager.go

```diff
@@ -0,0 +1,93 @@
+package sftp
+
+import (
+	stdjson "encoding/json"
+	"fmt"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/sftp"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/router/batchrouter/asyncdestinationmanager/common"
+)
+
+func (d *DefaultManager) Transform(job *jobsdb.JobT) (string, error) {
+	return common.GetMarshalledData(job, func(payload stdjson.RawMessage) string {
+		return string(payload)
+	})
+}
+
+// Upload uploads the data to the destination and marks all jobs to be completed
+func (d *DefaultManager) Upload(asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	destination := asyncDestStruct.Destination
+	textFilePath := asyncDestStruct.FileName
+	destinationID := destination.ID
+	destConfigJSON, err := json.Marshal(destination.Config)
+	if err != nil {
+		return generateErrorOutput(fmt.Sprintf("error marshalling destination config: %v", err.Error()), asyncDestStruct.ImportingJobIDs, destinationID)
+	}
+
+	result := gjson.ParseBytes(destConfigJSON)
+	uploadFilePath := result.Get("filePath").String()
+	uploadFilePath = getUploadFilePath(uploadFilePath)
+	fileFormat := result.Get("fileFormat").String()
+
+	// Generate temporary file based on the destination's file format
+	jsonOrCSVFilePath, err := generateFile(textFilePath, fileFormat)
+	if err != nil {
+		return generateErrorOutput(fmt.Sprintf("error generating temporary file: %v", err.Error()), asyncDestStruct.ImportingJobIDs, destinationID)
+	}
+
+	d.logger.Debugf("[Async Destination Manager] File Upload Started for Dest Type %v", destination.DestinationDefinition.Name)
```

Comment:
We can use non-sugared structured logging. More about it in [here](https://rudderlabs.slack.com/archives/C02LNLLAABC/p1708346761796859). Also, we can avoid cluttering with prefix `[Async Destination Manager]` because while creating the logger we are already defining it.
```suggestion
	d.logger.Debugn("File Upload Started", obskit.DestinationID(destinationID))
```

Reply:
Same of other logs as well.

Reply:
addressed